provider/manifest/manager.go

```diff
@@ -259,8 +260,25 @@ func (m *manager) maybeScheduleStop() bool { // nolint:golint,unparam
 	return true
 }
 
+func (m *manager) fillAllRequests(response error) {
+	for _, reqch := range m.pendingRequests {
+		reqch <- response
+	}
+	m.pendingRequests = nil
+
+	for _, req := range m.requests {
+		req.ch <- response
+	}
+	m.requests = nil
+}
+
 func (m *manager) emitReceivedEvents() {
-	if m.data == nil || len(m.leases) == 0 || len(m.manifests) == 0 {
+	if len(m.leases) == 0 {
+		m.log.Debug("emit received events skips due to no leases", "data", m.data, "leases", len(m.leases), "manifests", len(m.manifests))
+		m.fillAllRequests(ErrNoLeaseForDeployment)
+		return
```

Comment:
Again good catch.  The idea here was that there are cases where the manifest might be shared before the lease is obtained, but I think it's better that we shed all of these historical ideas and simplify as much as possible.  To that end, I'd be open to slimming down this service significantly - I've never been happy with how difficult it is to follow (and I wrote it).